<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />


<title>Identify gene expression or methylation features that can predict this factor</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMARTrial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Identify gene expression or methylation
features that can predict this factor</h1>
<h4 class="author">Junyan Lu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>SMARTrial/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-17
</li>
<li>
unnamed-chunk-19
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>featureSelection_cache</code> and re-run the analysis. To have
workflowr automatically delete the cache directory prior to building the
file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="load-libraries-and-datasets" class="section level1">
<h1>Load libraries and datasets</h1>
<p>Libraries</p>
<pre class="r"><code>library(jyluMisc)
library(glmnet)
library(SummarizedExperiment)
library(cowplot)
library(tidyverse)

options(warn=-1)
set.seed(2019)

ggplot2::theme_set(theme_bw() + theme(axis.text = element_text(size=15), axis.title = element_text(size=16), 
                                      plot.title = element_text(size=16, hjust = 0.5, face = &quot;bold&quot;),
                                      strip.text = element_text(size=16)))</code></pre>
</div>
<div id="prepare-dataset" class="section level1">
<h1>Prepare dataset</h1>
<div id="proteomics" class="section level2">
<h2>Proteomics</h2>
<p>Load dataset</p>
<pre class="r"><code>load(&quot;../output/protSmart_new.RData&quot;)
seProt &lt;- protSmart_raw[,protSmart_raw$DISEASE %in% &quot;AML&quot;]</code></pre>
<p>Remove proteins with high missing rate (50%)</p>
<pre class="r"><code>cut=0.5
seProt &lt;- seProt[rowMeans(is.na(assay(seProt))) &lt;= cut,]
dim(seProt)</code></pre>
<pre><code>[1] 8400   21</code></pre>
<p>VSN</p>
<pre class="r"><code>protMat &lt;- vsn::justvsn(assay(seProt))
assay(seProt) &lt;- protMat</code></pre>
<p>Prefilter by differential expression</p>
<pre class="r"><code>load(&quot;../output/resTab_proteomic_AML_CD34enriched.RData&quot;)
resTab &lt;- filter(resTab, pval &lt;= 0.05)

seProt &lt;- seProt[rownames(seProt) %in% resTab$name,]</code></pre>
<p>BPCA imputation</p>
<pre class="r"><code>seProt.imp &lt;- DEP::impute(seProt, &quot;bpca&quot;)</code></pre>
<p>Dimensions</p>
<pre class="r"><code>dim(seProt.imp)</code></pre>
<pre><code>[1] 640  21</code></pre>
</div>
<div id="metabolomics" class="section level2">
<h2>Metabolomics</h2>
<p>Load dataset</p>
<pre class="r"><code>load(&quot;../output/metaSmart.RData&quot;)
seMeta &lt;- metaSmart[,metaSmart$DISEASE == &quot;AML&quot; ]</code></pre>
<p>BPCA imputation</p>
<pre class="r"><code>seMeta.imp &lt;- DEP::impute(seMeta, &quot;bpca&quot;)</code></pre>
<p>Filter by differential abundance</p>
<pre class="r"><code>load(&quot;../output/resTab_metabolomic_AML_all.RData&quot;)
resTab &lt;- filter(resTab, pval &lt;= 0.1)
seMeta.imp &lt;- seMeta.imp[rownames(seMeta.imp) %in% resTab$name,]</code></pre>
<p>Dimensions</p>
<pre class="r"><code>dim(seMeta.imp)</code></pre>
<pre><code>[1] 32 21</code></pre>
</div>
<div id="drug-screens" class="section level2">
<h2>Drug screens</h2>
<pre class="r"><code>load(&quot;../output/seDrug.RData&quot;)
seDrug &lt;- seDrug_auc[,seDrug_auc$patientID %in% seProt$SMART_ID]
colnames(seDrug) &lt;- colnames(seProt[,match(colnames(seDrug), seProt$SMART_ID)])
#drugMat &lt;- jyluMisc::mscale(drugMat)</code></pre>
</div>
<div id="in-vivo-respones" class="section level2">
<h2>In-vivo respones</h2>
<pre class="r"><code>y &lt;- colData(seProt)[[&quot;Resp_manual&quot;]]
names(y) &lt;- colnames(seProt)
y &lt;- ifelse(!is.na(y), ifelse(y == &quot;R&quot;, 0, 1),NA)
y &lt;- y[!is.na(y)]</code></pre>
</div>
<div id="create-input-data-object" class="section level2">
<h2>Create input data object</h2>
<pre class="r"><code>allTrainData &lt;- list(
  protein = list(y = y, X = assay(seProt.imp)[,names(y)]),
  metabolite = list(y = y, X = assay(seMeta.imp)[,names(y)]),
  drug = list(y = y[intersect(names(y),colnames(seDrug))], X = assay(seDrug)[,intersect(names(y),colnames(seDrug))])
)</code></pre>
<p>Combined proteomic and metabolomic (not need to combine, protein will
always be selected over metabolites)</p>
<pre class="r"><code>#comMat &lt;- rbind(assay(seProt.imp),assay(seMeta.imp))
#allTrainData[[&quot;prot_meta&quot;]] &lt;- list(y = y, X = comMat[,names(y)])</code></pre>
</div>
</div>
<div id="use-random-lasso-to-rank-features-by-their-importance"
class="section level1">
<h1>Use random lasso to rank features by their importance</h1>
<pre class="r"><code>source(&quot;../code/Random_lasso.R&quot;)</code></pre>
<div id="calculate-feature-importance-patth-by-random-lasso"
class="section level2">
<h2>Calculate feature importance patth by random lasso</h2>
<pre class="r"><code>set.seed(2024)
rndLassoRes &lt;- list()
rndRes &lt;- lapply(names(allTrainData), function(n) {
  X &lt;- scale(t(allTrainData[[n]]$X))
  y &lt;- allTrainData[[n]]$y
  res &lt;- featurePath(X, y, sampleFraction = 0.8, weakness =0.2, nPerm = 1000, 
                        typePerm = &quot;standard&quot;, lambda = seq(1,0.01,length.out = 100), family = &quot;binomial&quot;)
  tibble(feature = rownames(res$freqMat),
         importance = rowMeans(res$freqMat),
         set = n)
}) %&gt;% bind_rows()</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="list-of-the-top-30-most-important-features"
class="section level2">
<h2>List of the top 30 most important features</h2>
<pre class="r"><code>pList &lt;- lapply(unique(rndRes$set), function(s) {
  eachTab &lt;- filter(rndRes, set == s) %&gt;%
    arrange(desc(importance)) %&gt;%
    slice_head(n=30) 
    if (s == &quot;protein&quot;) {
      eachTab &lt;- eachTab %&gt;% 
        mutate(feature = rowData(seProt[feature,])$Genes)
    }
    eachTab &lt;- mutate(eachTab, feature = factor(feature, levels = feature))
  ggplot(eachTab, aes(x=feature, y=importance)) +
    geom_bar(stat =&quot;identity&quot;) +
    ggtitle(s) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0),
          plot.title = element_text(face =&quot;bold&quot;, size=10, hjust = 0.5))
})
cowplot::plot_grid(plotlist= pList, ncol=1)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="test-the-performance-of-using-different-number-of-features"
class="section level1">
<h1>Test the performance of using different number of features</h1>
<pre class="r"><code>set.seed(2024)


featureNum &lt;- list(protein = seq(2,11, 1),
                   metabolite = seq(2,11, 1),
                   drug = seq(2,11, 1))

allCompareRes &lt;- lapply(unique(names(allTrainData)), function(n) {
    X &lt;- scale(t(allTrainData[[n]]$X))
    y &lt;- allTrainData[[n]]$y
    
    lapply(featureNum[[n]], function(i) {
      
       seleFeature &lt;- filter(rndRes, set == n) %&gt;%
        slice_max(importance, n=i) %&gt;% pull(feature)
      
       X.sub &lt;- X[,seleFeature,drop=FALSE]
       y.sub &lt;- y
       bootRes &lt;- evaluateModel(X.sub, y.sub, repeats = 100)
       sumBootResults(bootRes) %&gt;% mutate(set = n, n_feature = i)
    }) %&gt;% bind_rows()
}) %&gt;% bind_rows()</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div id="plot-the-results" class="section level2">
<h2>Plot the results</h2>
<pre class="r"><code>pList &lt;- lapply(unique(allCompareRes$name), function(metric) {
  plotTab &lt;- filter(allCompareRes, name == metric)
  ggplot(plotTab, aes(x=factor(n_feature), y=mean, fill = set)) +
    geom_bar(stat = &quot;identity&quot;, width = 0.8) +
    geom_errorbar(aes(ymax = upper.95, ymin = lower.95), width=0.5) +
    facet_wrap(~set, ncol=3, scale = &quot;free_x&quot;) +
    ggtitle(metric) +
    theme(legend.position = &quot;none&quot;) +
    ylab(&quot;value&quot;) + xlab(&quot;feature number&quot;)
})

cowplot::plot_grid(plotlist = pList, ncol=1)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-20-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-the-selected-10-proteins-as-heatmap"
class="section level2">
<h2>Visualize the selected 10 proteins as heatmap</h2>
<pre class="r"><code>seleFeature &lt;- filter(rndRes, set == &quot;protein&quot;) %&gt;%
      slice_max(importance, n=10) %&gt;% pull(feature)
plotMat &lt;- allTrainData$protein$X[seleFeature,]
rownames(plotMat) &lt;- rowData(seProt)[seleFeature,]$Genes
colAnno &lt;- tibble(id = colnames(seProt), group_3 = seProt$Resp_manual) %&gt;%
  filter(!is.na(group_3)) %&gt;%
  mutate(group_2 =ifelse(group_3==&quot;R&quot;,&quot;R&quot;,&quot;NR&quot;)) %&gt;%
  arrange(group_2) %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% data.frame()

plotMat &lt;- plotMat[,rownames(colAnno)]

pheatmap::pheatmap(plotMat, scale = &quot;row&quot;, cluster_cols = FALSE, annotation_col = colAnno)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-21-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-the-selected-7-metabolites-as-heatmap"
class="section level2">
<h2>Visualize the selected 7 metabolites as heatmap</h2>
<pre class="r"><code>seleFeature &lt;- filter(rndRes, set == &quot;metabolite&quot;) %&gt;%
      slice_max(importance, n=7) %&gt;% pull(feature)
plotMat &lt;- allTrainData$metabolite$X[seleFeature,]
colAnno &lt;- tibble(id = colnames(seMeta), group_3 = seMeta$Resp_manual) %&gt;%
  filter(!is.na(group_3)) %&gt;%
  mutate(group_2 =ifelse(group_3==&quot;R&quot;,&quot;R&quot;,&quot;NR&quot;)) %&gt;%
  arrange(group_2) %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% data.frame()

plotMat &lt;- plotMat[,rownames(colAnno)]

pheatmap::pheatmap(plotMat, scale = &quot;row&quot;, cluster_cols = FALSE, annotation_col = colAnno)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-22-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-the-selected-5-drugs-as-heatmap"
class="section level2">
<h2>Visualize the selected 5 drugs as heatmap</h2>
<pre class="r"><code>seleFeature &lt;- filter(rndRes, set == &quot;drug&quot;) %&gt;%
      slice_max(importance, n=7) %&gt;% pull(feature)
plotMat &lt;- allTrainData$drug$X[seleFeature,]
colAnno &lt;- tibble(id = colnames(seDrug), 
                  group_3 = seProt[,match(colnames(seDrug),colnames(seProt))]$Resp_manual) %&gt;%
  filter(!is.na(group_3)) %&gt;%
  mutate(group_2 =ifelse(group_3==&quot;R&quot;,&quot;R&quot;,&quot;NR&quot;)) %&gt;%
  arrange(group_2) %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% data.frame()

plotMat &lt;- plotMat[,rownames(colAnno)]

pheatmap::pheatmap(plotMat, scale = &quot;row&quot;, cluster_cols = FALSE, annotation_col = colAnno)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-23-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="scatter-plot-using-the-top2-proteins" class="section level2">
<h2>Scatter plot using the top2 proteins</h2>
<pre class="r"><code>seleFeature &lt;- filter(rndRes, set == &quot;protein&quot;) %&gt;%
      slice_max(importance, n=2) %&gt;% pull(feature)
protMat &lt;- allTrainData$protein$X[seleFeature,]
rownames(protMat) &lt;- rowData(seProt)[seleFeature,]$Genes

colAnno &lt;- tibble(id = colnames(seProt), group_3 = seProt$Resp_manual) %&gt;%
  filter(!is.na(group_3)) %&gt;%
  mutate(group_2 =ifelse(group_3==&quot;R&quot;,&quot;R&quot;,&quot;NR&quot;)) %&gt;%
  arrange(group_2) %&gt;%
  column_to_rownames(&quot;id&quot;) %&gt;% data.frame()

protTab &lt;- t(protMat) %&gt;%
  as_tibble(rownames = &quot;id&quot;) %&gt;%
  mutate(response = colAnno[id,]$group_2)
ggplot(protTab, aes(x=LLGL1, y = TTPAL)) +
  geom_point(aes(col = response),size=3)</code></pre>
<p><img src="figure/featureSelection.Rmd/unnamed-chunk-24-1.png" width="480" style="display: block; margin: auto;" />
LLGL1 regulates drug resistance of gemcitabine: <a
href="https://www.sciencedirect.com/science/article/pii/S2352345X20300989?via%3Dihub"
class="uri">https://www.sciencedirect.com/science/article/pii/S2352345X20300989?via%3Dihub</a><br />
TTPAL is an oncogene: <a
href="https://pubmed.ncbi.nlm.nih.gov/34642500/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/34642500/</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.1              
 [3] dplyr_1.1.4.9000            purrr_0.3.4                
 [5] readr_2.1.2                 tidyr_1.2.0                
 [7] tibble_3.2.1                ggplot2_3.4.1              
 [9] tidyverse_1.3.2             cowplot_1.1.1              
[11] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[13] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        
[15] IRanges_2.30.0              S4Vectors_0.34.0           
[17] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[19] matrixStats_0.62.0          glmnet_4.1-4               
[21] Matrix_1.5-4                jyluMisc_0.1.5             

loaded via a namespace (and not attached):
  [1] DEP_1.18.0             utf8_1.2.4             shinydashboard_0.7.2  
  [4] gmm_1.6-6              tidyselect_1.2.1       htmlwidgets_1.5.4     
  [7] grid_4.2.0             BiocParallel_1.30.3    norm_1.0-10.0         
 [10] maxstat_0.7-25         munsell_0.5.0          codetools_0.2-18      
 [13] preprocessCore_1.58.0  DT_0.23                withr_3.0.0           
 [16] colorspace_2.0-3       highr_0.9              knitr_1.39            
 [19] rstudioapi_0.13        ggsignif_0.6.3         mzID_1.34.0           
 [22] labeling_0.4.2         git2r_0.30.1           slam_0.1-50           
 [25] GenomeInfoDbData_1.2.8 KMsurv_0.1-5           pheatmap_1.0.12       
 [28] farver_2.1.1           rprojroot_2.0.3        vctrs_0.6.5           
 [31] generics_0.1.3         TH.data_1.1-1          xfun_0.31             
 [34] sets_1.0-21            R6_2.5.1               doParallel_1.0.17     
 [37] clue_0.3-61            MsCoreUtils_1.8.0      bitops_1.0-7          
 [40] cachem_1.0.6           fgsea_1.22.0           DelayedArray_0.22.0   
 [43] assertthat_0.2.1       promises_1.2.0.1       scales_1.2.0          
 [46] multcomp_1.4-19        googlesheets4_1.0.0    gtable_0.3.0          
 [49] affy_1.74.0            sandwich_3.0-2         workflowr_1.7.0       
 [52] rlang_1.1.3            mzR_2.30.0             GlobalOptions_0.1.2   
 [55] splines_4.2.0          rstatix_0.7.0          gargle_1.2.0          
 [58] impute_1.70.0          broom_1.0.0            BiocManager_1.30.18   
 [61] yaml_2.3.5             abind_1.4-5            modelr_0.1.8          
 [64] backports_1.4.1        httpuv_1.6.6           tools_4.2.0           
 [67] relations_0.6-12       affyio_1.66.0          ellipsis_0.3.2        
 [70] gplots_3.1.3           jquerylib_0.1.4        RColorBrewer_1.1-3    
 [73] MSnbase_2.22.0         plyr_1.8.7             Rcpp_1.0.9            
 [76] visNetwork_2.1.0       zlibbioc_1.42.0        RCurl_1.98-1.7        
 [79] ggpubr_0.4.0           GetoptLong_1.0.5       zoo_1.8-10            
 [82] haven_2.5.0            cluster_2.1.3          exactRankTests_0.8-35 
 [85] fs_1.5.2               magrittr_2.0.3         data.table_1.14.8     
 [88] circlize_0.4.15        reprex_2.0.1           survminer_0.4.9       
 [91] pcaMethods_1.88.0      googledrive_2.0.0      mvtnorm_1.1-3         
 [94] ProtGenerics_1.28.0    hms_1.1.1              shinyjs_2.1.0         
 [97] mime_0.12              evaluate_0.15          xtable_1.8-4          
[100] XML_3.99-0.10          readxl_1.4.0           gridExtra_2.3         
[103] shape_1.4.6            compiler_4.2.0         KernSmooth_2.23-20    
[106] ncdf4_1.19             crayon_1.5.2           htmltools_0.5.4       
[109] later_1.3.0            tzdb_0.3.0             lubridate_1.8.0       
[112] DBI_1.1.3              dbplyr_2.2.1           ComplexHeatmap_2.12.1 
[115] tmvtnorm_1.5           MASS_7.3-58            car_3.1-0             
[118] cli_3.6.2              imputeLCMD_2.1         vsn_3.64.0            
[121] marray_1.74.0          parallel_4.2.0         igraph_1.3.4          
[124] pkgconfig_2.0.3        km.ci_0.5-6            piano_2.12.0          
[127] MALDIquant_1.21        xml2_1.3.3             foreach_1.5.2         
[130] bslib_0.4.1            XVector_0.36.0         drc_3.0-1             
[133] rvest_1.0.2            digest_0.6.30          rmarkdown_2.14        
[136] cellranger_1.1.0       fastmatch_1.1-3        survMisc_0.5.6        
[139] shiny_1.7.4            gtools_3.9.3           rjson_0.2.21          
[142] lifecycle_1.0.4        jsonlite_1.8.3         carData_3.0-5         
[145] limma_3.52.2           fansi_1.0.6            pillar_1.9.0          
[148] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[151] plotrix_3.8-2          survival_3.4-0         glue_1.7.0            
[154] png_0.1-7              iterators_1.0.14       stringi_1.7.8         
[157] sass_0.4.2             caTools_1.18.2        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
