<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Identify metabolomic markers for drug response in SmartTrail cohort using the metabolomic dataset</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMARTrial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Identify metabolomic markers for drug
response in SmartTrail cohort using the metabolomic dataset</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-05-29
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMARTrial/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
</div>
<div id="focused-analysis-on-aml" class="section level1">
<h1>Focused analysis on AML</h1>
<pre class="r"><code>metaSub &lt;- metaSmart[,metaSmart$DISEASE %in% &quot;AML&quot;]</code></pre>
<div id="pca" class="section level2">
<h2>PCA</h2>
<pre class="r"><code>exprMat &lt;- assays(metaSub)[[&quot;imputed&quot;]]

smpAnno &lt;- colData(metaSub) %&gt;% as_tibble() %&gt;%
  mutate(respGroup = ifelse(!is.na(Resp_manual), 
                            ifelse(Resp_manual == &quot;R&quot;, &quot;responder&quot;,&quot;non-responder&quot;), NA))

pcRes &lt;- prcomp(t(exprMat), scale. = FALSE, center = TRUE)

pcTab &lt;- pcRes$x[,1:10] %&gt;% 
    as_tibble(rownames = &quot;sampleID&quot;) %&gt;%
    left_join(smpAnno)</code></pre>
<pre><code>Joining with `by = join_by(sampleID)`</code></pre>
<p>PC1 versus PC2</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>PC3 versus PC4</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC3, y=PC4)) +
    geom_point(aes(col = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="test-associations-between-pca-and-metadata"
class="section level2">
<h2>Test associations between PCA and metadata</h2>
<pre class="r"><code>pcs &lt;- pcRes$x[,1:10] %&gt;% 
    as_tibble(rownames = &quot;sampleID&quot;)
metaTab &lt;- smpAnno %&gt;%
  select(sampleID, SEX, MATERIAL, DISEASE, ENRICHMENT, INITIAL_FREEZING, DEAD_CELL_REMOVAL, THAW_BATCH, 
         PERCENT_VIABILITY, tumor_infiltration, Pretreatment, NPM1:Trisomy.8q24, Treatment_type, treatment_spec,chemo_pat,
         Resp_manual, respGroup) %&gt;%
  mutate(across(NPM1:Trisomy.8q24, as.factor))

resTab &lt;- jyluMisc::testAssociation(pcs, metaTab, joinID = &quot;sampleID&quot;) %&gt;%
  filter(p&lt;0.05)
head(resTab)</code></pre>
<pre><code>  var1             var2            p     p.adj
1  PC1             IDH1 0.0009803125 0.1862594
2  PC1 INITIAL_FREEZING 0.0063493703 0.3984713
3 PC10              CAK 0.0092341784 0.3984713
4  PC3       THAW_BATCH 0.0100966280 0.3984713
5  PC6     Pretreatment 0.0113205667 0.3984713
6  PC3       ENRICHMENT 0.0125833035 0.3984713</code></pre>
<p>No PC associated with resposne group</p>
<p>Enrichment</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = ENRICHMENT)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Response</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = respGroup, shape = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression-with-cd34-enrichment"
class="section level2">
<h2>Differential expression (with CD34 enrichment)</h2>
<div id="preprocessing-data" class="section level3">
<h3>Preprocessing data</h3>
<p>Subset samples:<br />
PD and SD will be considered as non-responders (NR) and will be compared
to responders (R) This is different to the SMARTrial study, in which
only PD were compared to responders. If we only use R here, there are
not enough samples.</p>
<pre class="r"><code>#dataset
load(&quot;../output/metaSmart.RData&quot;)

metaSub &lt;- metaSmart[, metaSmart$DISEASE %in% &quot;AML&quot; &amp;
                           metaSmart$ENRICHMENT %in% &quot;CD34&quot; &amp;
                           !is.na(metaSmart$Resp_manual)]

metaSub$respGroup &lt;- ifelse(metaSub$Resp_manual ==&quot;R&quot;, &quot;R&quot;, &quot;NR&quot;)
metaSub$respGroup &lt;- factor(metaSub$respGroup, levels = c(&quot;R&quot;,&quot;NR&quot;))</code></pre>
<p>Filter out proteins with too many missing values</p>
<pre class="r"><code>x &lt;- assay(metaSub)
t1 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;R&quot;]) &gt;= 3
t2 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;NR&quot;]) &gt;= 3
keepProt &lt;- t1 &amp; t2
metaSub &lt;- metaSub[keepProt, ]
dim(metaSub)</code></pre>
<pre><code>[1] 477  15</code></pre>
<p>Number of samples</p>
<pre class="r"><code>table(metaSub$respGroup)</code></pre>
<pre><code>
 R NR 
 9  6 </code></pre>
</div>
<div id="differential-expression-using-proda" class="section level3">
<h3>Differential expression using proDA</h3>
<pre class="r"><code>exprMat &lt;- assays(metaSub)[[&quot;imputed&quot;]]
designMat &lt;- model.matrix(~respGroup, colData(metaSub))
fit &lt;- limma::lmFit(exprMat, designMat)
fit2 &lt;- limma::eBayes(fit)
resTab &lt;- limma::topTable(fit2, coef = &quot;respGroupNR&quot;, number = Inf) %&gt;%
  as_tibble(rownames = &quot;name&quot;) %&gt;%
  dplyr::rename(pval = P.Value, adj_pval = adj.P.Val) %&gt;%
  arrange(pval)

hist(resTab$pval)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save the table for MOFA filtering</p>
<pre class="r"><code>save(resTab, file = &quot;../output/resTab_metabolomic_AML_CD34enriched.RData&quot;)</code></pre>
</div>
<div id="list-of-significant-associations" class="section level3">
<h3>List of significant associations</h3>
<pre class="r"><code>filter(resTab, pval &lt;= 0.05) %&gt;% mutate(across(where(is.numeric), formatC, digits=2)) %&gt;%
  select(name,  pval, adj_pval, logFC) %&gt;%
  DT::datatable()</code></pre>
<pre><code>Warning: There was 1 warning in `mutate()`.
ℹ In argument: `across(where(is.numeric), formatC, digits = 2)`.
Caused by warning:
! The `...` argument of `across()` is deprecated as of dplyr 1.1.0.
Supply arguments directly to `.fns` through an anonymous function instead.

  # Previously
  across(a:b, mean, na.rm = TRUE)

  # Now
  across(a:b, \(x) mean(x, na.rm = TRUE))</code></pre>
<div id="htmlwidget-de200b75918365e265c7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-de200b75918365e265c7">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22"],["PI.17.0_18.1","PE.34.0","LPE.22.4","PA.16.0_18.3","C8","TG.14.0_34.0","PE.40.4","PE.P.18.0.16.0","LPI.18.1","LPE.20.1","LPG.16.0","Cer.d18.1.14.0","PE.P.18.1.20.4","LPC.26.0","PG.18.1_18.1","PG.16.0_22.2","PE.28.0","PA.18.1_18.3","PA.18.1_18.1","Cer.d18.1.26.0","TG.22.5_34.3","Spermidine"],["0.002","0.0047","0.0074","0.0077","0.012","0.014","0.015","0.018","0.02","0.021","0.024","0.027","0.029","0.029","0.033","0.036","0.037","0.04","0.042","0.048","0.049","0.05"],["0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73","0.73"],["-0.99","-0.78","1.1","0.75","0.91","1.1","0.79","-0.84","-0.72","0.48","  1","-0.65","-0.53","0.73","-0.57","0.58","0.79","0.66","-0.47","-0.66","0.65","-0.73"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>pval<\/th>\n      <th>adj_pval<\/th>\n      <th>logFC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="boxplot-of-top-associations" class="section level3">
<h3>Boxplot of top associations</h3>
<pre class="r"><code>pList &lt;- lapply(seq(9), function(i) {
  rec &lt;- resTab[i,]
  plotTab &lt;- tibble(expr = exprMat[rec$name,],
                    group = metaSub$respGroup)
  ggplot(plotTab, aes(x=group, y = expr)) +
    geom_boxplot(aes(fill = group), alpha=0.5) +
    ggbeeswarm::geom_beeswarm() +
    ggtitle(sprintf(&quot;%s (P=%s)&quot;,rec$name, formatC(rec$pval, digits = 2))) +
    theme_bw()
})
cowplot::plot_grid(plotlist = pList, ncol=3)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="differential-expression-with-or-without-cd34-enrichment"
class="section level2">
<h2>Differential expression (with or without CD34 enrichment)</h2>
<div id="preprocessing-data-1" class="section level3">
<h3>Preprocessing data</h3>
<p>Subset samples:<br />
PD and SD will be considered as non-responders (NR) and will be compared
to responders (R) This is different to the SMARTrial study, in which
only PD were compared to responders. If we only use R here, there are
not enough samples.</p>
<pre class="r"><code>#dataset
load(&quot;../output/metaSmart.RData&quot;)

metaSub &lt;- metaSmart[, metaSmart$DISEASE %in% &quot;AML&quot; &amp;
                           #metaSmart$ENRICHMENT %in% &quot;CD34&quot; &amp;
                           !is.na(metaSmart$Resp_manual)]

metaSub$respGroup &lt;- ifelse(metaSub$Resp_manual ==&quot;R&quot;, &quot;R&quot;, &quot;NR&quot;)
metaSub$respGroup &lt;- factor(metaSub$respGroup, levels = c(&quot;R&quot;,&quot;NR&quot;))</code></pre>
<p>Filter out proteins with too many missing values</p>
<pre class="r"><code>x &lt;- assay(metaSub)
t1 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;R&quot;]) &gt;= 3
t2 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;NR&quot;]) &gt;= 3
keepProt &lt;- t1 &amp; t2
metaSub &lt;- metaSub[keepProt, ]
dim(metaSub)</code></pre>
<pre><code>[1] 478  17</code></pre>
<p>Number of samples</p>
<pre class="r"><code>table(metaSub$respGroup)</code></pre>
<pre><code>
 R NR 
10  7 </code></pre>
</div>
<div id="differential-expression-using-proda-1" class="section level3">
<h3>Differential expression using proDA</h3>
<pre class="r"><code>exprMat &lt;- assays(metaSub)[[&quot;imputed&quot;]]
designMat &lt;- model.matrix(~respGroup, colData(metaSub))
fit &lt;- limma::lmFit(exprMat, designMat)
fit2 &lt;- limma::eBayes(fit)
resTab &lt;- limma::topTable(fit2, coef = &quot;respGroupNR&quot;, number = Inf) %&gt;%
  as_tibble(rownames = &quot;name&quot;) %&gt;%
  dplyr::rename(pval = P.Value, adj_pval = adj.P.Val) %&gt;%
  arrange(pval)

hist(resTab$pval)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save the table for MOFA filtering</p>
<pre class="r"><code>save(resTab, file = &quot;../output/resTab_metabolomic_AML_all.RData&quot;)</code></pre>
</div>
<div id="list-of-significant-associations-1" class="section level3">
<h3>List of significant associations</h3>
<pre class="r"><code>filter(resTab, pval &lt;= 0.05) %&gt;% mutate(across(where(is.numeric), formatC, digits=2)) %&gt;%
  select(name,  pval, adj_pval, logFC) %&gt;%
  DT::datatable()</code></pre>
<div id="htmlwidget-38d1d2f4af7240b5e321" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-38d1d2f4af7240b5e321">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["PE.34.0","TG.14.0_34.0","PI.17.0_18.1","PA.16.0_18.3","LPE.20.1","PE.P.20.0.20.5","C16","PE.P.18.0.16.0","TG.20.4_34.1","LPG.16.0","PE.32.0","Cer.d18.1.26.0","LPE.22.4","TG.16.0_35.2","PE.40.4"],["0.0008","0.012","0.015","0.017","0.019","0.02","0.022","0.024","0.026","0.03","0.031","0.034","0.034","0.04","0.048"],["0.38","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92"],["-0.9","0.96","-0.79","0.69","0.46","0.5","-0.85","-0.73","-1.1","  1","-0.61","-0.66","0.84","-0.94","0.61"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>pval<\/th>\n      <th>adj_pval<\/th>\n      <th>logFC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="boxplot-of-top-associations-1" class="section level3">
<h3>Boxplot of top associations</h3>
<pre class="r"><code>pList &lt;- lapply(seq(9), function(i) {
  rec &lt;- resTab[i,]
  plotTab &lt;- tibble(expr = exprMat[rec$name,],
                    group = metaSub$respGroup)
  ggplot(plotTab, aes(x=group, y = expr)) +
    geom_boxplot(aes(fill = group), alpha=0.5) +
    ggbeeswarm::geom_beeswarm() +
    ggtitle(sprintf(&quot;%s (P=%s)&quot;,rec$name, formatC(rec$pval, digits = 2))) +
    theme_bw()
})
cowplot::plot_grid(plotlist = pList, ncol=3)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="focused-analysis-all-cll" class="section level1">
<h1>Focused analysis all CLL</h1>
<pre class="r"><code>metaSub &lt;- metaSmart[,metaSmart$DISEASE %in% &quot;CLL&quot;]</code></pre>
<div id="pca-1" class="section level2">
<h2>PCA</h2>
<pre class="r"><code>exprMat &lt;- assays(metaSub)[[&quot;imputed&quot;]]

smpAnno &lt;- colData(metaSub) %&gt;% as_tibble() %&gt;%
  mutate(respGroup = ifelse(!is.na(Resp_manual), 
                            ifelse(Resp_manual == &quot;R&quot;, &quot;responder&quot;,&quot;non-responder&quot;), NA))

pcRes &lt;- prcomp(t(exprMat), scale. = FALSE, center = TRUE)

pcTab &lt;- pcRes$x[,1:10] %&gt;% 
    as_tibble(rownames = &quot;sampleID&quot;) %&gt;%
    left_join(smpAnno)</code></pre>
<pre><code>Joining with `by = join_by(sampleID)`</code></pre>
<p>PC1 versus PC2</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>PC3 versus PC4</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC3, y=PC4)) +
    geom_point(aes(col = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="test-associations-between-pca-and-metadata-1"
class="section level2">
<h2>Test associations between PCA and metadata</h2>
<pre class="r"><code>pcs &lt;- pcRes$x[,1:10] %&gt;% 
    as_tibble(rownames = &quot;sampleID&quot;)
metaTab &lt;- smpAnno %&gt;%
  select(sampleID, SEX, MATERIAL, DISEASE, ENRICHMENT, INITIAL_FREEZING, DEAD_CELL_REMOVAL, THAW_BATCH, 
         PERCENT_VIABILITY, tumor_infiltration, Pretreatment, NPM1:Trisomy.8q24, Treatment_type, treatment_spec,chemo_pat,
         Resp_manual, respGroup) %&gt;%
  mutate(across(NPM1:Trisomy.8q24, as.factor))

resTab &lt;- jyluMisc::testAssociation(pcs, metaTab, joinID = &quot;sampleID&quot;) %&gt;%
  filter(p&lt;0.05)
head(resTab)</code></pre>
<pre><code>  var1              var2           p     p.adj
1  PC3    Trisomie_12q13 0.001257624 0.1662400
2  PC6 DEAD_CELL_REMOVAL 0.001511273 0.1662400
3  PC1 PERCENT_VIABILITY 0.006246262 0.4580592
4  PC6        THAW_BATCH 0.009637405 0.4580981
5  PC5    Trisomie_12q13 0.013236211 0.4580981
6  PC9  INITIAL_FREEZING 0.013372520 0.4580981</code></pre>
<p>No PC associated with resposne group</p>
<p>Enrichment</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = ENRICHMENT)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Response</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2)) +
    geom_point(aes(col = respGroup, shape = Resp_manual)) +
    #ggrepel::geom_text_repel(aes(label = id)) +
    theme_bw()</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-expression" class="section level2">
<h2>Differential expression</h2>
<div id="preprocessing-data-2" class="section level3">
<h3>Preprocessing data</h3>
<p>Subset samples:<br />
Only use AML with chemotherapy and enrichned with CD34. PD and SD will
be considered as non-responders (NR) and will be compared to responders
(R) This is different to the SMARTrial study, in which only PD were
compared to responders. If we only use R here, there are not enough
samples.</p>
<pre class="r"><code>#dataset
load(&quot;../output/metaSmart.RData&quot;)

metaSub &lt;- metaSmart[, metaSmart$DISEASE %in% &quot;CLL&quot; &amp;
                           !is.na(metaSmart$Resp_manual)]

metaSub$respGroup &lt;- ifelse(metaSub$Resp_manual ==&quot;R&quot;, &quot;R&quot;, &quot;NR&quot;)
metaSub$respGroup &lt;- factor(metaSub$respGroup, levels = c(&quot;R&quot;,&quot;NR&quot;))</code></pre>
<p>Filter out proteins with too many missing values</p>
<pre class="r"><code>t1 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;R&quot;]) &gt;= 3
t2 &lt;- rowSums(!is.na(assay(metaSub))[,metaSub$respGroup==&quot;NR&quot;]) &gt;= 3
keepProt &lt;- t1 &amp; t2
metaSub &lt;- metaSub[keepProt, ]
dim(metaSub)</code></pre>
<pre><code>[1] 478  24</code></pre>
<p>Number of samples</p>
<pre class="r"><code>table(metaSub$respGroup)</code></pre>
<pre><code>
 R NR 
19  5 </code></pre>
</div>
<div id="differential-expression-using-limma" class="section level3">
<h3>Differential expression using limma</h3>
<pre class="r"><code>exprMat &lt;- assays(metaSub)[[&quot;imputed&quot;]]
designMat &lt;- model.matrix(~respGroup, colData(metaSub))
fit &lt;- limma::lmFit(exprMat, designMat)
fit2 &lt;- limma::eBayes(fit)
resTab &lt;- limma::topTable(fit2, coef = &quot;respGroupNR&quot;, number = Inf) %&gt;%
  as_tibble(rownames = &quot;name&quot;) %&gt;%
  dplyr::rename(pval = P.Value, adj_pval = adj.P.Val) %&gt;%
  arrange(pval)

hist(resTab$pval)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="list-of-significant-associations-2" class="section level3">
<h3>List of significant associations</h3>
<pre class="r"><code>filter(resTab, pval &lt;= 0.05) %&gt;% mutate(across(where(is.numeric), formatC, digits=2)) %&gt;%
  select(name,  pval, adj_pval, logFC) %&gt;%
  DT::datatable()</code></pre>
<div id="htmlwidget-93a7a9d76886afece8d9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-93a7a9d76886afece8d9">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11"],["PA.18.2_22.1","SPB.d16.1","LPE.16.1","Leu","LPS.18.1","MG.20.4","PE.44.6","Choline","PG.18.2_20.5","SM.35.1","LPE.20.2"],["0.0033","0.0086","0.012","0.015","0.016","0.017","0.026","0.027","0.036","0.047","0.048"],["0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92","0.92"],["0.61","-0.54","0.61","-0.69","-0.74"," -1","0.82","-0.8","0.43","-0.39","0.55"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>pval<\/th>\n      <th>adj_pval<\/th>\n      <th>logFC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="boxplot-of-top-associations-2" class="section level3">
<h3>Boxplot of top associations</h3>
<pre class="r"><code>pList &lt;- lapply(seq(9), function(i) {
  rec &lt;- resTab[i,]
  plotTab &lt;- tibble(expr = exprMat[rec$name,],
                    group = metaSub$respGroup)
  ggplot(plotTab, aes(x=group, y = expr)) +
    geom_boxplot(aes(fill = group), alpha=0.5) +
    ggbeeswarm::geom_beeswarm() +
    ggtitle(sprintf(&quot;%s (P=%s)&quot;,rec$name, formatC(rec$pval, digits = 2))) +
    theme_bw()
})
cowplot::plot_grid(plotlist = pList, ncol=3)</code></pre>
<p><img src="figure/metabolomicAnalysis.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.1              
 [3] dplyr_1.1.4.9000            purrr_0.3.4                
 [5] readr_2.1.2                 tidyr_1.2.0                
 [7] tibble_3.2.1                ggplot2_3.4.1              
 [9] tidyverse_1.3.2             SummarizedExperiment_1.26.1
[11] Biobase_2.56.0              GenomicRanges_1.48.0       
[13] GenomeInfoDb_1.32.2         IRanges_2.30.0             
[15] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[17] MatrixGenerics_1.8.1        matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] readxl_1.4.0           backports_1.4.1        fastmatch_1.1-3       
  [4] drc_3.0-1              jyluMisc_0.1.5         workflowr_1.7.0       
  [7] igraph_1.3.4           shinydashboard_0.7.2   splines_4.2.0         
 [10] crosstalk_1.2.0        BiocParallel_1.30.3    TH.data_1.1-1         
 [13] digest_0.6.30          htmltools_0.5.4        fansi_1.0.6           
 [16] magrittr_2.0.3         googlesheets4_1.0.0    cluster_2.1.3         
 [19] tzdb_0.3.0             limma_3.52.2           modelr_0.1.8          
 [22] sandwich_3.0-2         piano_2.12.0           colorspace_2.0-3      
 [25] rvest_1.0.2            haven_2.5.0            xfun_0.31             
 [28] crayon_1.5.2           RCurl_1.98-1.7         jsonlite_1.8.3        
 [31] survival_3.4-0         zoo_1.8-10             glue_1.7.0            
 [34] survminer_0.4.9        gtable_0.3.0           gargle_1.2.0          
 [37] zlibbioc_1.42.0        XVector_0.36.0         DelayedArray_0.22.0   
 [40] car_3.1-0              abind_1.4-5            scales_1.2.0          
 [43] mvtnorm_1.1-3          DBI_1.1.3              relations_0.6-12      
 [46] rstatix_0.7.0          Rcpp_1.0.9             plotrix_3.8-2         
 [49] xtable_1.8-4           km.ci_0.5-6            DT_0.23               
 [52] htmlwidgets_1.5.4      httr_1.4.3             fgsea_1.22.0          
 [55] gplots_3.1.3           ellipsis_0.3.2         pkgconfig_2.0.3       
 [58] farver_2.1.1           sass_0.4.2             dbplyr_2.2.1          
 [61] utf8_1.2.4             tidyselect_1.2.1       labeling_0.4.2        
 [64] rlang_1.1.3            later_1.3.0            munsell_0.5.0         
 [67] cellranger_1.1.0       tools_4.2.0            visNetwork_2.1.0      
 [70] cachem_1.0.6           cli_3.6.2              generics_0.1.3        
 [73] broom_1.0.0            evaluate_0.15          fastmap_1.1.0         
 [76] yaml_2.3.5             knitr_1.39             fs_1.5.2              
 [79] survMisc_0.5.6         caTools_1.18.2         mime_0.12             
 [82] slam_0.1-50            xml2_1.3.3             compiler_4.2.0        
 [85] rstudioapi_0.13        beeswarm_0.4.0         ggsignif_0.6.3        
 [88] marray_1.74.0          reprex_2.0.1           bslib_0.4.1           
 [91] stringi_1.7.8          highr_0.9              lattice_0.20-45       
 [94] Matrix_1.5-4           KMsurv_0.1-5           shinyjs_2.1.0         
 [97] vctrs_0.6.5            pillar_1.9.0           lifecycle_1.0.4       
[100] jquerylib_0.1.4        data.table_1.14.8      cowplot_1.1.1         
[103] bitops_1.0-7           httpuv_1.6.6           R6_2.5.1              
[106] promises_1.2.0.1       KernSmooth_2.23-20     gridExtra_2.3         
[109] vipor_0.4.5            codetools_0.2-18       MASS_7.3-58           
[112] gtools_3.9.3           exactRankTests_0.8-35  assertthat_0.2.1      
[115] rprojroot_2.0.3        withr_3.0.0            multcomp_1.4-19       
[118] GenomeInfoDbData_1.2.8 parallel_4.2.0         hms_1.1.1             
[121] grid_4.2.0             rmarkdown_2.14         carData_3.0-5         
[124] googledrive_2.0.0      ggpubr_0.4.0           git2r_0.30.1          
[127] maxstat_0.7-25         sets_1.0-21            shiny_1.7.4           
[130] lubridate_1.8.0        ggbeeswarm_0.6.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
