<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />


<title>Integrative analysis using MOFA for SmartTrail dataset (only AML sample and semi-supervised)</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMARTrial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Integrative analysis using MOFA for
SmartTrail dataset (only AML sample and semi-supervised)</h1>
<h4 class="author">Junyan Lu</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMARTrial/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
</div>
<div id="data-preprocessing" class="section level1">
<h1>Data preprocessing</h1>
<div id="proteomics" class="section level2">
<h2>Proteomics</h2>
<p>Load dataset</p>
<pre class="r"><code>load(&quot;../output/protSmart_new.RData&quot;)
protSmart_raw &lt;- protSmart_raw[,protSmart_raw$DISEASE %in% &quot;AML&quot;]</code></pre>
<p>Remove proteins with high missing rate (50%)</p>
<pre class="r"><code>cut=0.5
protData_filter &lt;- protSmart_raw[rowSums(is.na(assay(protSmart_raw)))/ncol(protSmart_raw) &lt;= cut,]
dim(protData_filter)</code></pre>
<pre><code>[1] 8400   21</code></pre>
<p>Remove proteins without any symbols</p>
<pre class="r"><code>protData_filter &lt;- protData_filter[!rowData(protData_filter)$Genes %in% c(NA,&quot;&quot;),]</code></pre>
<p>VSN</p>
<pre class="r"><code>protMat &lt;- vsn::justvsn(assay(protData_filter))</code></pre>
<p>Change rownames to protein names</p>
<pre class="r"><code>#use symbol as row names, may introduce duplicates
rownames(protMat) &lt;- getOneSymbol(rowData(protSmart_raw[rownames(protMat),])$Genes)

#order protein by sds
sds &lt;- apply(protMat, 1, function(x) sd(x, na.rm=TRUE))
protMat &lt;- protMat[order(sds, decreasing = TRUE),]

#remove duplicates
protMat &lt;- protMat[!duplicated(rownames(protMat)),]</code></pre>
<p>Prefilter by differential expression</p>
<pre class="r"><code>load(&quot;../output/resTab_proteomic_AML.RData&quot;)
resTab &lt;- filter(resTab, pval &lt;= 0.05)

protMat &lt;- protMat[rownames(protMat) %in% resTab$symbol,]</code></pre>
<p>Dimensions</p>
<pre class="r"><code>dim(protMat)</code></pre>
<pre><code>[1] 481  21</code></pre>
</div>
<div id="metabolomics" class="section level2">
<h2>Metabolomics</h2>
<p>Load dataset</p>
<pre class="r"><code>load(&quot;../output/metaSmart.RData&quot;)
metaSmart &lt;- metaSmart[,metaSmart$DISEASE == &quot;AML&quot; ]</code></pre>
<p>Get metabolite abundance matrix, no futher processing is
necessary</p>
<pre class="r"><code>metaMat &lt;- assay(metaSmart)</code></pre>
<p>Filter by differential abundance</p>
<pre class="r"><code>load(&quot;../output/resTab_metabolomic_AML.RData&quot;)
resTab &lt;- filter(resTab, pval &lt;= 0.05)
metaMat &lt;- metaMat[rownames(metaMat) %in% resTab$name,]</code></pre>
<p>Dimensions</p>
<pre class="r"><code>dim(metaMat)</code></pre>
<pre><code>[1] 15 21</code></pre>
</div>
<div id="drug-screens" class="section level2">
<h2>Drug screens</h2>
<pre class="r"><code>load(&quot;../output/seDrug.RData&quot;)
drugMat &lt;- assay(seDrug_auc[,seDrug_auc$patientID %in% protData_filter$SMART_ID])
colnames(drugMat) &lt;- colnames(protSmart_raw[,match(colnames(drugMat), protSmart_raw$SMART_ID)])
#drugMat &lt;- jyluMisc::mscale(drugMat)</code></pre>
<p>Prefilter by differential sensitivity</p>
<pre class="r"><code>patTab &lt;- colData(protSmart)[colnames(drugMat),]
patTab$respGroup &lt;- ifelse(!is.na(patTab$Resp_protocol), ifelse(patTab$Resp_protocol == &quot;R&quot;, &quot;R&quot;, &quot;NR&quot;),NA)
designMat &lt;- model.matrix(~respGroup, patTab)
testMat &lt;- drugMat[,rownames(designMat)]
fit &lt;- limma::lmFit(testMat, designMat)
fit2 &lt;- limma::eBayes(fit)
resTab &lt;- limma::topTable(fit2, number=Inf)</code></pre>
<p>No drug passed 0.05 p-value, drug data will not be filtered</p>
<pre class="r"><code>filter(resTab, P.Value &lt; 0.05)</code></pre>
<pre><code>[1] logFC     AveExpr   t         P.Value   adj.P.Val B        
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
<div id="get-useful-metadata-information" class="section level2">
<h2>Get useful metadata information</h2>
<pre class="r"><code>seleCol &lt;- str_split(str_remove_all(&quot;SEX, tumor_infiltration, age, MATERIAL, DISEASE, INITIAL_FREEZING, ENRICHMENT, DEAD_CELL_REMOVAL, THAW_BATCH,PERCENT_VIABILITY, CELL_NUMBER_SUBMITTED..x10.6., Treatment_type, Resp_protocol, Resp_manual,Pretreatment, NPM1, CEBPA, FLT3, FLT3_ITD_ratio, IDH1, IGHV.status, chemo_pat,BIRC3,MLL_rearragement, CAK, Del17p, Del.11q, Del.13q, Trisomie_12q13, Trisomy.8q24,time_OS,  Death, time_EFS&quot;,&quot; &quot;), 
                     &quot;,&quot;)[[1]]
patMeta &lt;- colData(protSmart_raw)[,seleCol]
patMeta$respGroup &lt;- ifelse(!is.na(patMeta$Resp_protocol), ifelse(patMeta$Resp_protocol == &quot;R&quot;, &quot;R&quot;, &quot;NR&quot;),NA)
patMeta$respGroup_manual &lt;- ifelse(!is.na(patMeta$Resp_manual), ifelse(patMeta$Resp_manual == &quot;R&quot;, &quot;R&quot;, &quot;NR&quot;),NA)</code></pre>
</div>
</div>
<div id="create-mofa-object" class="section level1">
<h1>Create MOFA object</h1>
<pre class="r"><code>mofaData &lt;- list(Protein = protMat, 
                 Metabolite = metaMat,
                 Drug = drugMat)

# Create MultiAssayExperiment object 
mofaData &lt;- MultiAssayExperiment::MultiAssayExperiment(
  experiments = mofaData,
  colData = patMeta
)</code></pre>
<pre class="r"><code>MOFAobject &lt;- create_mofa_from_MultiAssayExperiment(mofaData, extract_metadata = TRUE)</code></pre>
<p>Plot data overview</p>
<pre class="r"><code>plot_data_overview(MOFAobject)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="running-mofa-model" class="section level1">
<h1>Running MOFA model</h1>
<pre class="r"><code>running_mofa &lt;- function(obj, var = 0.01, mode = &quot;fast&quot;, nFac =20, 
                         maxIter=10000, 
                         spikeslab_factors = FALSE, 
                         spikeslab_weights= FALSE){
  
  ### Setup MOFA training parameters
  # Define data options
  DataOptions &lt;- get_default_data_options(obj)
  DataOptions$use_float32 &lt;- FALSE

  #Define model options
  ModelOptions &lt;- get_default_model_options(obj)
  #ModelOptions$likelihoods[&quot;Gene&quot;] &lt;- &quot;bernoulli&quot;
  # number of factors to start with
  # (in each iteration number of factors is tried to be reduced)
  ModelOptions$num_factors &lt;- nFac
  ModelOptions$spikeslab_factors &lt;- spikeslab_factors
  ModelOptions$spikeslab_weights &lt;- spikeslab_weights

  #ModelOptions

  # Define training options
  TrainOptions &lt;- get_default_training_options(obj)
  TrainOptions$convergence_mode &lt;- mode
  TrainOptions$maxiter &lt;- maxIter
  
  # Automatically drop factors that explain less than x% of variance
  TrainOptions$drop_factor_threshold &lt;- var
  #TrainOptions
  
  
  ### Run MOFA model
  obj_prep &lt;- prepare_mofa(
    obj, 
    data_options = DataOptions,
    model_options = ModelOptions,
    training_options = TrainOptions
  )

  obj_res &lt;- run_mofa(obj_prep, use_basilisk = FALSE)
  return(obj_res)
}</code></pre>
<pre class="r"><code>set.seed(2023)
MOFAobject_res &lt;- running_mofa(obj=MOFAobject, var=0.01, mode =&quot;slow&quot;, nFac = 10)
save(MOFAobject_res, file = &quot;../output/mofaRes_AML_filt.RData&quot;)</code></pre>
</div>
<div
id="explore-mofa-factors-to-identify-factors-associated-with-resistance"
class="section level1">
<h1>Explore MOFA factors to identify factors associated with
resistance</h1>
<p>Load save results</p>
<pre class="r"><code>load(&quot;../output/mofaRes_AML_filt.RData&quot;)</code></pre>
<div id="explor-mofa-model" class="section level2">
<h2>Explor MOFA model</h2>
<div id="variance-explained-for-each-view" class="section level3">
<h3>Variance explained for each view</h3>
<pre class="r"><code>plot_variance_explained(MOFAobject_res, x=&quot;view&quot;, y=&quot;factor&quot;)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="factor-correlation-matrix" class="section level3">
<h3>Factor correlation matrix</h3>
<pre class="r"><code>plot_factor_cor(MOFAobject_res)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-23-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="total-variance-explained" class="section level3">
<h3>Total variance explained</h3>
<pre class="r"><code>plot_variance_explained(MOFAobject_res, plot_total = T)[[2]]</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-24-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="factor-association-with-metadata" class="section level3">
<h3>Factor association with metadata</h3>
<pre class="r"><code>res &lt;- correlate_factors_with_covariates(MOFAobject_res, alpha = 0.05, adjustP = FALSE, return_data = FALSE)</code></pre>
<img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/MOFA_correlation_metadata-1.png" width="768" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was
ignored by workflowr.
</div>
<pre class="r"><code>#res &lt;- res[,!colnames(res) %in% c(&quot;group&quot;,&quot;sample&quot;)]
#pheatmap::pheatmap(res)</code></pre>
<p><strong>It seems Factor 6 may be associated with in-vivo
resistance</strong></p>
</div>
<div id="factor-1-which-may-be-associated-with-in-vivo-response"
class="section level3">
<h3>Factor 1, which may be associated with in-vivo response</h3>
<pre class="r"><code>nFac = 1</code></pre>
<div id="factor-scores" class="section level4">
<h4>Factor scores</h4>
<pre class="r"><code>plot_factor(MOFAobject_res, factors=nFac, color_by = &quot;Resp_protocol&quot;, dodge = TRUE, add_boxplot = TRUE)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="factor-weights" class="section level4">
<h4>Factor weights</h4>
<p>Proteomics</p>
<pre class="r"><code>plot_data_heatmap(MOFAobject_res, factor = nFac, view = &quot;Protein&quot;, 
                  annotation_samples = c(&quot;Resp_protocol&quot;), 
                  features = 20, cluster_rows = FALSE, cluster_cols = FALSE, scale= &quot;row&quot;)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Metabolites</p>
<pre class="r"><code>plot_data_heatmap(MOFAobject_res, factor = nFac, view = &quot;Metabolite&quot;, 
                  annotation_samples = c(&quot;Resp_protocol&quot;,  &quot;NPM1&quot;, &quot;FLT3&quot;),
                  features = 20, cluster_rows = FALSE, cluster_cols = FALSE, scale= &quot;row&quot;)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Drugs</p>
<pre class="r"><code>plot_data_heatmap(MOFAobject_res, factor = nFac, view = &quot;Drug&quot;, 
                  annotation_samples = c(&quot;Resp_protocol&quot;,  &quot;NPM1&quot;, &quot;FLT3&quot;),
                  features = 20, cluster_rows = FALSE, cluster_cols = FALSE, scale= &quot;row&quot;)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
</div>
<div
id="further-characterization-of-factor-6-which-may-associate-with-in-vivo-chemotherapy-resistance"
class="section level1">
<h1>Further characterization of Factor 6, which may associate with
in-vivo chemotherapy resistance</h1>
<pre class="r"><code>facTab &lt;- get_factors(MOFAobject_res)[[1]] %&gt;%
    as_tibble(rownames=&quot;patID&quot;) %&gt;% 
    left_join(as_tibble(patMeta, rownames = &quot;patID&quot;) , 
              by = &quot;patID&quot;)</code></pre>
<div id="boxplots" class="section level2">
<h2>Boxplots</h2>
<div id="binary-definition" class="section level3">
<h3>Binary definition</h3>
<p>Resp_protocl</p>
<pre class="r"><code>ggplot(facTab, aes(x=respGroup, y=Factor1)) +
  geom_boxplot() + geom_point(aes(col = respGroup)) +
  geom_text(aes(label = patID))</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Resp_manual (manually added some in-vivo response by JLU)</p>
<pre class="r"><code>ggplot(facTab, aes(x=respGroup_manual, y=Factor1)) +
  geom_boxplot() + geom_point(aes(col = respGroup_manual)) +
  geom_text(aes(label = patID))</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="three-groups" class="section level3">
<h3>Three groups</h3>
<p>Resp_protocol</p>
<pre class="r"><code>ggplot(facTab, aes(x=Resp_protocol, y=Factor1)) +
  geom_boxplot() + geom_point(aes(col = Resp_protocol))</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-34-1.png" width="672" style="display: block; margin: auto;" />
<strong>Higher Factor 6 values are associated with more
resistance</strong></p>
<p>Resp_manual</p>
<pre class="r"><code>ggplot(facTab, aes(x=Resp_manual, y=Factor1)) +
  geom_boxplot() + geom_point(aes(col = Resp_manual))</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-35-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="biological-intepretation-of-factor-6" class="section level2">
<h2>Biological intepretation of Factor 6</h2>
<pre class="r"><code>nFac = 1</code></pre>
<div id="weight-of-proteomics-features" class="section level3">
<h3>Weight of proteomics features</h3>
<pre class="r"><code>plot_top_weights(MOFAobject_res,
 view = &quot;Protein&quot;,
 factor = nFac,
 nfeatures = 20,     # Top number of features to highlight
 scale = T           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-37-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="pathway-enrichment-analysis" class="section level3">
<h3>Pathway Enrichment analysis</h3>
<div id="canonical-pathways" class="section level4">
<h4>Canonical pathways</h4>
<pre class="r"><code>#prepare genesets
cpSet &lt;- piano::loadGSC(&quot;~/CLLproject_jlu/data/commonFiles/c2.cp.v6.2.symbols.gmt&quot;)$gsc

sigMat &lt;- lapply(names(cpSet), function(x) {
    data.frame(name = x, gene = cpSet[[x]])
}) %&gt;% bind_rows() %&gt;% mutate(val =1) %&gt;%
    pivot_wider(values_from = val, names_from = gene) %&gt;%
    column_to_rownames(&quot;name&quot;) %&gt;% as.matrix()
sigMat[is.na(sigMat)] &lt;- 0

#negative
enrichment.neg &lt;- run_enrichment(MOFAobject_res,
  view = &quot;Protein&quot;, factors = nFac,
  feature.sets = sigMat,
  sign = &quot;negative&quot;,
  statistical.test = &quot;parametric&quot;
)

#positive
enrichment.pos &lt;- run_enrichment(MOFAobject_res,
  view = &quot;Protein&quot;, factors = nFac,
  feature.sets = sigMat,
  sign = &quot;positive&quot;,
  statistical.test = &quot;parametric&quot;
)</code></pre>
<div id="positively-associated-no-pathways-found"
class="section level5">
<h5>Positively associated (no pathways found)</h5>
<pre class="r"><code>#plot_enrichment(enrichment.pos, 
#  factor = 1, 
#  max.pathways = 20
#)</code></pre>
</div>
<div id="negatively-associated" class="section level5">
<h5>Negatively associated</h5>
<pre class="r"><code>plot_enrichment(enrichment.neg, 
  factor = 1, 
  max.pathways = 20
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-40-1.png" width="1152" style="display: block; margin: auto;" />
#### GO Biological processes</p>
<pre class="r"><code>#prepare genesets
cpSet &lt;- piano::loadGSC(&quot;~/CLLproject_jlu/data/commonFiles/c5.bp.v6.2.symbols.gmt&quot;)$gsc

sigMat &lt;- lapply(names(cpSet), function(x) {
    data.frame(name = x, gene = cpSet[[x]])
}) %&gt;% bind_rows() %&gt;% mutate(val =1) %&gt;%
    pivot_wider(values_from = val, names_from = gene) %&gt;%
    column_to_rownames(&quot;name&quot;) %&gt;% as.matrix()
sigMat[is.na(sigMat)] &lt;- 0

#negative
enrichment.neg &lt;- run_enrichment(MOFAobject_res,
  view = &quot;Protein&quot;, factors = nFac,
  feature.sets = sigMat,
  sign = &quot;negative&quot;,
  statistical.test = &quot;parametric&quot;
)

#positive
enrichment.pos &lt;- run_enrichment(MOFAobject_res,
  view = &quot;Protein&quot;, factors = nFac,
  feature.sets = sigMat,
  sign = &quot;positive&quot;,
  statistical.test = &quot;parametric&quot;
)</code></pre>
</div>
<div id="positively-associated" class="section level5">
<h5>Positively associated</h5>
<pre class="r"><code>plot_enrichment(enrichment.pos, 
  factor = 1, 
  max.pathways = 20
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-42-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="negatively-associated-1" class="section level5">
<h5>Negatively associated</h5>
<pre class="r"><code>plot_enrichment(enrichment.neg, 
  factor = 1, 
  max.pathways = 20
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-43-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="weight-of-metabolomic-features" class="section level3">
<h3>Weight of metabolomic features</h3>
<pre class="r"><code>plot_top_weights(MOFAobject_res,
 view = &quot;Metabolite&quot;,
 factor = nFac,
 nfeatures = 20,     # Top number of features to highlight
 scale = T           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-44-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="weight-of-drug-features" class="section level3">
<h3>Weight of drug features</h3>
<pre class="r"><code>plot_top_weights(MOFAobject_res,
 view = &quot;Drug&quot;,
 factor = nFac,
 nfeatures = 20,     # Top number of features to highlight
 scale = T           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/mofaAnalysis_onlyAML_prefilter.Rmd/unnamed-chunk-45-1.png" width="768" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.1              
 [3] dplyr_1.1.4.9000            purrr_0.3.4                
 [5] readr_2.1.2                 tidyr_1.2.0                
 [7] tibble_3.2.1                ggplot2_3.4.1              
 [9] tidyverse_1.3.2             MOFA2_1.6.0                
[11] MultiAssayExperiment_1.22.0 SummarizedExperiment_1.26.1
[13] Biobase_2.56.0              GenomicRanges_1.48.0       
[15] GenomeInfoDb_1.32.2         IRanges_2.30.0             
[17] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[19] MatrixGenerics_1.8.1        matrixStats_0.62.0         
[21] jyluMisc_0.1.5             

loaded via a namespace (and not attached):
  [1] utf8_1.2.4             shinydashboard_0.7.2   reticulate_1.25       
  [4] tidyselect_1.2.1       htmlwidgets_1.5.4      grid_4.2.0            
  [7] BiocParallel_1.30.3    Rtsne_0.16             maxstat_0.7-25        
 [10] munsell_0.5.0          preprocessCore_1.58.0  codetools_0.2-18      
 [13] DT_0.23                withr_3.0.0            colorspace_2.0-3      
 [16] filelock_1.0.2         highr_0.9              knitr_1.39            
 [19] rstudioapi_0.13        ggsignif_0.6.3         labeling_0.4.2        
 [22] git2r_0.30.1           slam_0.1-50            GenomeInfoDbData_1.2.8
 [25] KMsurv_0.1-5           farver_2.1.1           pheatmap_1.0.12       
 [28] rhdf5_2.40.0           rprojroot_2.0.3        basilisk_1.8.0        
 [31] vctrs_0.6.5            generics_0.1.3         TH.data_1.1-1         
 [34] xfun_0.31              sets_1.0-21            R6_2.5.1              
 [37] bitops_1.0-7           rhdf5filters_1.8.0     cachem_1.0.6          
 [40] fgsea_1.22.0           DelayedArray_0.22.0    assertthat_0.2.1      
 [43] promises_1.2.0.1       scales_1.2.0           multcomp_1.4-19       
 [46] googlesheets4_1.0.0    gtable_0.3.0           affy_1.74.0           
 [49] sandwich_3.0-2         workflowr_1.7.0        rlang_1.1.3           
 [52] splines_4.2.0          rstatix_0.7.0          gargle_1.2.0          
 [55] broom_1.0.0            BiocManager_1.30.18    yaml_2.3.5            
 [58] reshape2_1.4.4         abind_1.4-5            modelr_0.1.8          
 [61] backports_1.4.1        httpuv_1.6.6           tools_4.2.0           
 [64] relations_0.6-12       affyio_1.66.0          ellipsis_0.3.2        
 [67] gplots_3.1.3           jquerylib_0.1.4        RColorBrewer_1.1-3    
 [70] Rcpp_1.0.9             plyr_1.8.7             visNetwork_2.1.0      
 [73] zlibbioc_1.42.0        RCurl_1.98-1.7         basilisk.utils_1.8.0  
 [76] ggpubr_0.4.0           cowplot_1.1.1          zoo_1.8-10            
 [79] haven_2.5.0            ggrepel_0.9.1          cluster_2.1.3         
 [82] exactRankTests_0.8-35  fs_1.5.2               magrittr_2.0.3        
 [85] data.table_1.14.8      reprex_2.0.1           survminer_0.4.9       
 [88] googledrive_2.0.0      mvtnorm_1.1-3          hms_1.1.1             
 [91] shinyjs_2.1.0          mime_0.12              evaluate_0.15         
 [94] xtable_1.8-4           readxl_1.4.0           gridExtra_2.3         
 [97] compiler_4.2.0         KernSmooth_2.23-20     crayon_1.5.2          
[100] htmltools_0.5.4        later_1.3.0            tzdb_0.3.0            
[103] lubridate_1.8.0        DBI_1.1.3              corrplot_0.92         
[106] dbplyr_2.2.1           MASS_7.3-58            Matrix_1.5-4          
[109] car_3.1-0              cli_3.6.2              vsn_3.64.0            
[112] marray_1.74.0          parallel_4.2.0         igraph_1.3.4          
[115] pkgconfig_2.0.3        km.ci_0.5-6            dir.expiry_1.4.0      
[118] piano_2.12.0           xml2_1.3.3             bslib_0.4.1           
[121] XVector_0.36.0         drc_3.0-1              rvest_1.0.2           
[124] digest_0.6.30          rmarkdown_2.14         cellranger_1.1.0      
[127] fastmatch_1.1-3        survMisc_0.5.6         uwot_0.1.11           
[130] shiny_1.7.4            gtools_3.9.3           lifecycle_1.0.4       
[133] jsonlite_1.8.3         Rhdf5lib_1.18.2        carData_3.0-5         
[136] limma_3.52.2           fansi_1.0.6            pillar_1.9.0          
[139] lattice_0.20-45        fastmap_1.1.0          httr_1.4.3            
[142] plotrix_3.8-2          survival_3.4-0         glue_1.7.0            
[145] png_0.1-7              stringi_1.7.8          sass_0.4.2            
[148] HDF5Array_1.24.1       caTools_1.18.2        </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
