<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Preprocess SMARTrial drug screen data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SMARTrial</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preprocess SMARTrial drug screen data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SMARTrial/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220425code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220425)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220425code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220425)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
</div>
<div id="load-and-preprocess-drug-screen-data" class="section level1">
<h1>Load and preprocess drug screen data</h1>
<div id="match-ids" class="section level2">
<h2>Match IDs</h2>
<pre class="r"><code>load(&quot;../data/smartData.RData&quot;)
smartScreen &lt;- mutate_if(smartScreen, is.factor, as.character)
#get overlapped samples
overPat &lt;- intersect(protSmart$SMART_ID, smartScreen$patientID)</code></pre>
<p>proteomic samples that do not have drug screen data</p>
<pre class="r"><code>#proteomic samples that do not have drug screen data
smps &lt;- protSmart$SMART_ID[! protSmart$SMART_ID %in% overPat]
smps</code></pre>
<pre><code>[1] &quot;21SCPB0009&quot; &quot;20SCPB0159&quot; &quot;20SCPB0157&quot; &quot;20SCPB0017&quot; &quot;21LN0303&quot;  
[6] &quot;19SCPB0167&quot;</code></pre>
<p>Those are the samples only have drug screen data.</p>
<p>Get sample IDs for drug screen samples</p>
<pre class="r"><code>smartScreen &lt;- separate(smartScreen, sampleID, into = c(&quot;a&quot;,&quot;smpID&quot;,&quot;b&quot;), sep = &quot;_&quot;, remove = FALSE)
smps %in% smartScreen$smpID</code></pre>
<pre><code>[1] TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<p>Now all the samples can be found.</p>
<p>For those 5 samples, change their SMART_ID (PatientID) to
sampleID</p>
<pre class="r"><code>smartScreen &lt;- smartScreen %&gt;% mutate(patientID = ifelse(smpID %in% smps, smpID, patientID))</code></pre>
<p>Wether all the proteomic samples have screen data?</p>
<pre class="r"><code>#get overlapped samples
all(protSmart$SMART_ID %in% smartScreen$patientID)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>screenData &lt;- filter(smartScreen, patientID %in% protSmart$SMART_ID) %&gt;%
  select(-a,-b,-smpID)</code></pre>
</div>
<div id="quality-control-of-drug-screen-data" class="section level2">
<h2>Quality control of drug screen data</h2>
<div id="calculate-sd-of-inner-negative-controls-per-plate"
class="section level3">
<h3>Calculate SD of inner negative controls per-plate</h3>
<pre class="r"><code>negPlate &lt;- filter(screenData, Drug == &quot;DMSO&quot;, !str_detect(well, &quot;^[ABOP]|(01|02|23|24)$&quot;))
sdNeg &lt;- group_by(negPlate, plateID) %&gt;% summarise(sdVal = sd(normVal))
meanNeg &lt;- group_by(negPlate, plateID) %&gt;% summarise(meanVal = mean(val))</code></pre>
</div>
<div
id="for-duplicated-patient-samples-keep-the-one-from-the-plate-with-less-sd-of-inner-negative-controls"
class="section level3">
<h3>For duplicated patient samples, keep the one from the plate with
less sd of inner negative controls</h3>
<pre class="r"><code>viabTab &lt;- mutate(screenData, 
                  innerNegSd = sdNeg[match(plateID, sdNeg$plateID),]$sdVal,
                  baseATP = meanNeg[match(plateID, meanNeg$plateID),]$meanVal) %&gt;%
  arrange(innerNegSd) %&gt;%
  filter(Drug!= &quot;DMSO&quot;) %&gt;%
  distinct(patientID, Drug, Concentration, .keep_all = TRUE) %&gt;%
  mutate(drugConc = paste0(Drug,&quot;_&quot;,Concentration)) </code></pre>
<pre class="r"><code>plotTab &lt;- distinct(viabTab, plateID, .keep_all = TRUE) %&gt;%
  mutate(Disease = protSmart[,match(patientID,protSmart$SMART_ID)]$DISEASE) %&gt;% as_tibble()
ggplot(plotTab, aes(x=innerNegSd, y=log(baseATP))) +
  geom_point(aes(color = Disease)) +
  geom_smooth() +
  ggrepel::geom_text_repel(aes(label= patientID))</code></pre>
<pre><code>`geom_smooth()` using method = &#39;loess&#39; and formula = &#39;y ~ x&#39;</code></pre>
<pre><code>Warning: ggrepel: 81 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pca" class="section level3">
<h3>PCA</h3>
<pre class="r"><code>viabMat &lt;- viabTab %&gt;% 
  select(patientID, drugConc, normVal) %&gt;%
  pivot_wider(names_from = patientID, values_from = normVal) %&gt;%
  column_to_rownames(&quot;drugConc&quot;) %&gt;% as.matrix()</code></pre>
<div id="evaluate-the-completeness" class="section level4">
<h4>Evaluate the completeness</h4>
<pre class="r"><code>naMat &lt;- viabMat
naMat[is.na(viabMat)] &lt;- 1
naMat[!is.na(viabMat)] &lt;- 0
pheatmap::pheatmap(naMat)</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Remove drug that have too many missing values</p>
<pre class="r"><code>viabMat &lt;- viabMat[rowMeans(naMat) &lt; 0.5, ]
viabTab &lt;- filter(viabTab, drugConc %in% rownames(viabMat))</code></pre>
<p>Only use the complete measurement for PCA</p>
<pre class="r"><code>viabMat.complete &lt;- viabMat[,complete.cases(t(viabMat))] </code></pre>
</div>
<div id="perform-pca" class="section level4">
<h4>Perform PCA</h4>
<pre class="r"><code>pcRes &lt;- prcomp(t(viabMat.complete),center = TRUE, scale. = TRUE)

sdNegAnno &lt;- distinct(viabTab, patientID, innerNegSd) %&gt;% group_by(patientID) %&gt;% summarise(innerNegSd = mean(innerNegSd))
baseATPAnno &lt;- distinct(viabTab, patientID, baseATP) %&gt;% group_by(patientID) %&gt;% summarise(baseATP = mean(baseATP))

pcTab &lt;- pcRes$x[,1:20] %&gt;%
  as_tibble(rownames = &quot;SMART_ID&quot;) %&gt;%
  left_join(colData(protSmart) %&gt;% as_tibble()) %&gt;%
  left_join(sdNegAnno, by = c(SMART_ID = &quot;patientID&quot;)) %&gt;%
  left_join(baseATPAnno, by = c(SMART_ID = &quot;patientID&quot;))</code></pre>
<pre><code>Joining, by = &quot;SMART_ID&quot;</code></pre>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2, color = DISEASE)) +
  geom_point(aes(size = innerNegSd)) +
  ggrepel::geom_text_repel(aes(label = SMART_ID))</code></pre>
<pre><code>Warning: ggrepel: 9 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="does-pc1-correlate-with-overall-viability"
class="section level5">
<h5>Does PC1 correlate with overall viability?</h5>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=log(baseATP))) +
  geom_point() +
  geom_smooth(method =  &quot;lm&quot;)</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>S007 looks like an outlier in PCA, is a samples with inner negative
control SD &gt; 0.15, better to remove it.</p>
<pre class="r"><code>viabTab &lt;- filter(viabTab, patientID != &quot;S007&quot;)
viabMat.complete &lt;- viabMat.complete[,colnames(viabMat.complete) %in% viabTab$patientID]</code></pre>
</div>
</div>
<div id="perform-pca-without-outlier" class="section level4">
<h4>Perform PCA without outlier</h4>
<pre class="r"><code>pcRes &lt;- prcomp(t(viabMat.complete),center = TRUE, scale. = TRUE)

sdNegAnno &lt;- distinct(viabTab, patientID, innerNegSd) %&gt;% group_by(patientID) %&gt;% summarise(innerNegSd = mean(innerNegSd))
baseATPAnno &lt;- distinct(viabTab, patientID, baseATP) %&gt;% group_by(patientID) %&gt;% summarise(baseATP = mean(baseATP))

pcTab &lt;- pcRes$x[,1:20] %&gt;%
  as_tibble(rownames = &quot;SMART_ID&quot;) %&gt;%
  left_join(colData(protSmart) %&gt;% as_tibble()) %&gt;%
  left_join(sdNegAnno, by = c(SMART_ID = &quot;patientID&quot;)) %&gt;%
  left_join(baseATPAnno, by = c(SMART_ID = &quot;patientID&quot;))</code></pre>
<pre><code>Joining, by = &quot;SMART_ID&quot;</code></pre>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=PC2, color = DISEASE)) +
  geom_point(aes(size = innerNegSd)) +
  ggrepel::geom_text_repel(aes(label = SMART_ID))</code></pre>
<pre><code>Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" />
##### Does PC1 correlate with overall viability?</p>
<pre class="r"><code>ggplot(pcTab, aes(x=PC1, y=log(baseATP))) +
  geom_point() +
  geom_smooth(method =  &quot;lm&quot;)</code></pre>
<pre><code>`geom_smooth()` using formula = &#39;y ~ x&#39;</code></pre>
<p><img src="figure/processDrugScreen.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="generate-summarised-assay" class="section level4">
<h4>Generate summarised assay</h4>
<pre class="r"><code>#Per-concentration

seDrug_conc &lt;- jyluMisc::tidyToSum(viabTab, rowID = &quot;drugConc&quot;, colID = &quot;patientID&quot;, values = c(&quot;normVal&quot;, &quot;normVal.cor&quot;),
                                   annoRow = c(&quot;Drug&quot;,&quot;Concentration&quot;, &quot;drugConc&quot;, &quot;concIndex&quot;,&quot;Target&quot;,&quot;Pathway&quot;),
                                   annoCol = c(&quot;patientID&quot;))
viabTab.auc &lt;- distinct(viabTab, patientID, Drug, .keep_all = TRUE)

#AUC
seDrug_auc &lt;- jyluMisc::tidyToSum(viabTab.auc, rowID = &quot;Drug&quot;, colID = &quot;patientID&quot;, values = c(&quot;normVal_auc&quot;, &quot;normVal.cor_auc&quot;),
                                   annoRow = c(&quot;Drug&quot;,&quot;Target&quot;,&quot;Pathway&quot;),
                                   annoCol = c(&quot;patientID&quot;))
save(seDrug_auc, seDrug_conc, viabTab, file = &quot;../output/seDrug.RData&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.1               stringr_1.4.1              
 [3] dplyr_1.0.9                 purrr_0.3.4                
 [5] readr_2.1.2                 tidyr_1.2.0                
 [7] tibble_3.1.8                ggplot2_3.4.1              
 [9] tidyverse_1.3.2             SummarizedExperiment_1.26.1
[11] Biobase_2.56.0              GenomicRanges_1.48.0       
[13] GenomeInfoDb_1.32.2         IRanges_2.30.0             
[15] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[17] MatrixGenerics_1.8.1        matrixStats_0.62.0         

loaded via a namespace (and not attached):
  [1] readxl_1.4.0           backports_1.4.1        fastmatch_1.1-3       
  [4] drc_3.0-1              jyluMisc_0.1.5         workflowr_1.7.0       
  [7] igraph_1.3.4           shinydashboard_0.7.2   splines_4.2.0         
 [10] BiocParallel_1.30.3    TH.data_1.1-1          digest_0.6.30         
 [13] htmltools_0.5.4        fansi_1.0.3            magrittr_2.0.3        
 [16] googlesheets4_1.0.0    cluster_2.1.3          tzdb_0.3.0            
 [19] limma_3.52.2           modelr_0.1.8           sandwich_3.0-2        
 [22] piano_2.12.0           colorspace_2.0-3       rvest_1.0.2           
 [25] ggrepel_0.9.1          haven_2.5.0            xfun_0.31             
 [28] crayon_1.5.2           RCurl_1.98-1.7         jsonlite_1.8.3        
 [31] survival_3.4-0         zoo_1.8-10             glue_1.6.2            
 [34] survminer_0.4.9        gtable_0.3.0           gargle_1.2.0          
 [37] zlibbioc_1.42.0        XVector_0.36.0         DelayedArray_0.22.0   
 [40] car_3.1-0              abind_1.4-5            scales_1.2.0          
 [43] pheatmap_1.0.12        mvtnorm_1.1-3          DBI_1.1.3             
 [46] relations_0.6-12       rstatix_0.7.0          Rcpp_1.0.9            
 [49] plotrix_3.8-2          xtable_1.8-4           km.ci_0.5-6           
 [52] DT_0.23                htmlwidgets_1.5.4      httr_1.4.3            
 [55] fgsea_1.22.0           gplots_3.1.3           RColorBrewer_1.1-3    
 [58] ellipsis_0.3.2         pkgconfig_2.0.3        farver_2.1.1          
 [61] sass_0.4.2             dbplyr_2.2.1           utf8_1.2.2            
 [64] tidyselect_1.1.2       labeling_0.4.2         rlang_1.0.6           
 [67] later_1.3.0            visNetwork_2.1.0       munsell_0.5.0         
 [70] cellranger_1.1.0       tools_4.2.0            cachem_1.0.6          
 [73] cli_3.4.1              generics_0.1.3         broom_1.0.0           
 [76] evaluate_0.15          fastmap_1.1.0          yaml_2.3.5            
 [79] knitr_1.39             fs_1.5.2               survMisc_0.5.6        
 [82] caTools_1.18.2         nlme_3.1-158           mime_0.12             
 [85] slam_0.1-50            xml2_1.3.3             compiler_4.2.0        
 [88] rstudioapi_0.13        ggsignif_0.6.3         marray_1.74.0         
 [91] reprex_2.0.1           bslib_0.4.1            stringi_1.7.8         
 [94] highr_0.9              lattice_0.20-45        Matrix_1.5-4          
 [97] KMsurv_0.1-5           shinyjs_2.1.0          vctrs_0.5.2           
[100] pillar_1.8.0           lifecycle_1.0.3        jquerylib_0.1.4       
[103] data.table_1.14.8      cowplot_1.1.1          bitops_1.0-7          
[106] httpuv_1.6.6           R6_2.5.1               promises_1.2.0.1      
[109] KernSmooth_2.23-20     gridExtra_2.3          codetools_0.2-18      
[112] MASS_7.3-58            gtools_3.9.3           exactRankTests_0.8-35 
[115] assertthat_0.2.1       rprojroot_2.0.3        withr_2.5.0           
[118] multcomp_1.4-19        GenomeInfoDbData_1.2.8 parallel_4.2.0        
[121] mgcv_1.8-40            hms_1.1.1              grid_4.2.0            
[124] rmarkdown_2.14         carData_3.0-5          googledrive_2.0.0     
[127] ggpubr_0.4.0           git2r_0.30.1           maxstat_0.7-25        
[130] sets_1.0-21            shiny_1.7.4            lubridate_1.8.0       </code></pre>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
